sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -244,6 +246,92 @@ case class HashAggregateExec(
 
   protected override val shouldStopRequired = false
 
+  // We assume a prefix has lower cases and a name has camel cases
+  private val variableName = "^[a-z]+_[a-zA-Z]+[0-9]*".r
+
+  // Returns true if a given name id belongs to this `CodegenContext`
+  private def isVariable(nameId: String): Boolean = nameId match {
+    case variableName() => true
+    case _ => false
+  }
+
+  // Extracts all the outer references for a given `aggExpr`. This result will be used to split
+  // aggregation into small functions.
+  private def getOuterReferences(
```

Comment:
`OuterReference` actually has special meaning in correlated subquery. This name can be confusing.

Reply:
ok, I'll rename this.